File: solutions/cpp/armstrong-numbers/armstrong_numbers.cpp
#include "armstrong_numbers.h"

namespace armstrong_numbers {

	bool is_armstrong_number(unsigned int num) {
		std::string s = std::to_string(num);
		unsigned int total = 0;

		for (auto& c : s)
			total += pow(double(c) - '0', s.size());
		
		return total == num;
	}

}  // namespace armstrong_numbers

File: solutions/java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {

        String s = Integer.toString(numberToCheck);
        int exp = s.length();

        for(int i = 0; i < exp; i++)
            numberToCheck -= Math.pow(s.charAt(i) - '0', exp);        

        return numberToCheck == 0;
    }

}
